File: Leetcode/11 - Container With Most Water/solution.ts
function maxArea(height: number[]): number {
    let start: number = 0;
    let end: number = height.length - 1;
    let maxContainer: number = 0;

    while (start < end) {
        const h: number = Math.min(height[start], height[end]);
        const width: number = end - start;
        maxContainer = Math.max(maxContainer, h * width);

        if (height[start] < height[end]) {
            start++;
        } else {
            end--;
        }
    }

    return maxContainer;
};

File: Leetcode/11 - Container With Most Water/solution.py
def maxArea(self, height: List[int]) -> int:
    start: int = 0
    end: int = len(height) - 1
    max_container: int = 0

    while start < end:
        h: int = min(height[start], height[end])
        width: int = end - start
        max_container = max(max_container, h * width)

        if height[start] < height[end]:
            start += 1
        else:
            end -= 1

    return max_container